src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -279,31 +281,36 @@ public UpdateResult applyTripUpdates(
           tripDescriptor
         );
 
-        Result<UpdateSuccess, UpdateError> result =
-          switch (tripScheduleRelationship) {
-            case SCHEDULED -> handleScheduledTrip(
-              tripUpdate,
-              tripId,
-              serviceDate,
-              backwardsDelayPropagationType
-            );
-            case ADDED -> validateAndHandleAddedTrip(
-              tripUpdate,
-              tripDescriptor,
-              tripId,
-              serviceDate
-            );
-            case CANCELED -> handleCanceledTrip(tripId, serviceDate, CancelationType.CANCEL);
-            case DELETED -> handleCanceledTrip(tripId, serviceDate, CancelationType.DELETE);
-            case REPLACEMENT -> validateAndHandleModifiedTrip(
-              tripUpdate,
-              tripDescriptor,
-              tripId,
-              serviceDate
-            );
-            case UNSCHEDULED -> UpdateError.result(tripId, NOT_IMPLEMENTED_UNSCHEDULED);
-            case DUPLICATED -> UpdateError.result(tripId, NOT_IMPLEMENTED_DUPLICATED);
-          };
+        Result<UpdateSuccess, UpdateError> result;
+        try {
+          result =
+            switch (tripScheduleRelationship) {
+              case SCHEDULED -> handleScheduledTrip(
+                tripUpdate,
+                tripId,
+                serviceDate,
+                backwardsDelayPropagationType
+              );
+              case ADDED -> validateAndHandleAddedTrip(
+                tripUpdate,
+                tripDescriptor,
+                tripId,
+                serviceDate
+              );
+              case CANCELED -> handleCanceledTrip(tripId, serviceDate, CancelationType.CANCEL);
+              case DELETED -> handleCanceledTrip(tripId, serviceDate, CancelationType.DELETE);
+              case REPLACEMENT -> validateAndHandleModifiedTrip(
+                tripUpdate,
+                tripDescriptor,
+                tripId,
+                serviceDate
+              );
+              case UNSCHEDULED -> UpdateError.result(tripId, NOT_IMPLEMENTED_UNSCHEDULED);
+              case DUPLICATED -> UpdateError.result(tripId, NOT_IMPLEMENTED_DUPLICATED);
+            };
+        } catch (DataValidationException e) {
+          result = DataValidationExceptionMapper.toResult(e);
+        }
```

Comment:
This is already fairly complicated. Can you move all of this into a method which returns the `Result`?

Reply:
I am not too familiar with the class, but it obvious need some care. We have plans to fix it, until then I would like to change as little as possible. I agree this would be a candidate for a local method, but hopefully we can split framework code and business logic a bit better. Is this the most important piece of code in this method - I kind of get that feeling, in that case every thing else should be hidden away and this should remain...

Reply:
Ok, lets keep it. I can take a look at it when I have time for it.